allensdk/brain_observatory/behavior/behavior_project_cache.py

```diff
@@ -0,0 +1,371 @@
+            },
+        }
+
+    def __init__(
+            self,
+            fetch_api: BehaviorProjectApi = BehaviorProjectLimsApi.default(),
+            fetch_tries: int = 2,
+            **kwargs):
+        """ Entrypoint for accessing visual behavior data. Supports
+        access to summaries of session data and provides tools for
+        downloading detailed session data (such as dff traces).
+
+        --- NOTE ---
+        Because NWB files are not currently supported for this project (as of
+        11/2019), this cache will not actually save any files of session data
+        to the local machine. Only summary tables will be saved to the local
+        cache. File retrievals for specific sessions will be handled by
+        the fetch api used for the Session object, and cached in-memory
+        only to enable fast retrieval for subsequent calls.
+
+        If you are looping over session objects, be sure to clean up
+        your memory when it is not needed by calling `cache_clear` from
+        your session object.
+
+        Parameters
+        ==========
+        fetch_api :
+            Used to pull data from remote sources, after which it is locally
+            cached. Any object inheriting from BehaviorProjectBase is
+            suitable. Current options are:
+                EcephysProjectLimsApi :: Fetches bleeding-edge data from the
+                    Allen Institute"s internal database. Only works if you are
+                    on our internal network.
+        fetch_tries :
+            Maximum number of times to attempt a download before giving up and
+            raising an exception. Note that this is total tries, not retries
+        **kwargs :
+            manifest : str or Path
+                full path at which manifest json will be stored
+            version : str
+                version of manifest file. If this mismatches the version
+                recorded in the file at manifest, an error will be raised.
+            other kwargs are passed to allensdk.api.cache.Cache
+        """
+        kwargs["manifest"] = kwargs.get("manifest",
+                                        "behavior_project_manifest.json")
+        kwargs["version"] = kwargs.get("version", self.MANIFEST_VERSION)
+
+        super().__init__(**kwargs)
+        self.fetch_api = fetch_api
+        self.fetch_tries = fetch_tries
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    @classmethod
+    def from_lims(cls, lims_kwargs=None, **kwargs):
+        lims_kwargs_ = lims_kwargs or {}
+        return cls(fetch_api=BehaviorProjectLimsApi.default(**lims_kwargs_),
+                   **kwargs)
+
+    def get_session_table(
+            self,
+            suppress: Optional[List[str]] = None,
+            by: str = "ophys_session_id") -> pd.DataFrame:
+        """
+        Return summary table of all ophys_session_ids in the database.
+        :param suppress: optional list of columns to drop from the resulting
```

Comment:
AllenSDK mostly uses numpy style docstrings. Any reason not to here?

Reply:
Just my thinking that we needed sphinx style, noted for the future!